### ISSUE TITLE
Cannot run multi-objective optimization after upgrading storage from v2.3.0 to master.

### ISSUE THREAD

**Issue by toshihikoyanase** [optuna/optuna#2179]
## Expected behavior

When I upgrade the Optuna storage in MySQL 8.0 from v2.3.0 to the current master, I couldn't execute the multi-objective optimization due to the integrity error of `trial_id`. The single objective optimization can be executed even after the upgrade, so I think the unique constraint of `trial_id` is not set appropriately. 

Optimization should be executed after upgrading the storage.

## Environment

- Optuna version: v2.3.0 to master
- Python version: 3.7.7
- OS: macOS 10.15.7
- (Optional) Other libraries and their versions:
  - MySQL==8.0.22
  - PyMySQL==0.9.3

## Error messages, stack traces, or logs

```console
sqlalchemy.exc.IntegrityError: (pymysql.err.IntegrityError) (1062, "Duplicate entry '4' for key 'trial_values.trial_id'")
[SQL: INSERT INTO trial_values (trial_id, objective, value) VALUES (%(trial_id)s, %(objective)s, %(value)s)]
[parameters: {'trial_id': 4, 'objective': 1, 'value': 0.18631845259151747}]
```

## Steps to reproduce

1. Launch MySQL server
2. Run optimization with optuna==v2.3.0
3. Upgrade storage using optuna@master
4. Run multi-objective optimization with optuna@master

## Reproducible examples (optional)

1. Launch MySQL server
```console
$ docker run --name mysql -e MYSQL_ROOT_PASSWORD=test -p 3306:3306 -p 33060:33060 -d mysql:8.0.22
$ docker run --network host -it --rm mysql:8.0.22  mysql -h 127.0.0.1 -uroot -ptest -e "create database migration_optuna;"
```

2. Run optimization with optuna==v2.3.0
```python
import optuna

def objective(trial):
    return trial.suggest_float("x", 0, 1)

study = optuna.create_study(study_name="single", storage="mysql+pymysql://root:test@localhost:3306/migration_optuna", load_if_exists=True)
study.optimize(objective, n_trials=1)
```

3. Upgrade storage using optuna@master
```console
$ optuna storage upgrade --storage mysql+pymysql://root:test@localhost:3306/migration_optuna
```

4. Run multi-objective optimization with optuna@master
```python
import optuna

def objective(trial):
    return trial.suggest_float("x", 0, 1), trial.suggest_float("y", 0, 1)

study = optuna.create_study(directions=["maximize", "minimize"], study_name="bi", storage="mysql+pymysql://root:test@localhost:3306/migration_optuna", load_if_exists=True)
study.optimize(objective, n_trials=1)
```

## Additional context (optional)

The trial_values table of upgraded RDB has `trial_id` as a unique key in addition to the combined unique key of `trial_id` and `objective`.

>   UNIQUE KEY `trial_id` (`trial_id`),
>   UNIQUE KEY `uq_trial_values_trial_id_objective` (`trial_id`,`objective`),

This is because the upgrade script simply adds the combined unique constraint without removing the existing unique constraint. I guess we need to remove it.

https://github.com/optuna/optuna/blob/add10e9fc5d734915c1cbe2d872db25c8297837c/optuna/storages/_rdb/alembic/versions/v2.4.0.a.py#L139-L146

**Comment by toshihikoyanase** [comment]
If you have already upgraded the storage, you can remove the inappropriate unique constraint by executing the following SQL:

```sql
alter table trial_values drop index trial_id;
```

Please make sure to backup your database before executing it.

**Comment by hvy** [comment]
Let me close this issue as fixed since #2180 has been merged.
